components/src/dynamo/planner/core/perf_model/rust_adapter.py

```
@@ -0,0 +1,714 @@
+try:  # pragma: no cover - availability depends on the optional Rust feature.
+    from dynamo.llm import (
+        AicEngineConfig,
+        EngineCapacityRequest,
+        EnginePerfLimits,
+        OptimizationTarget,
+        RustEnginePerfModel,
+        RustEnginePerfOptions,
+    )
+
+    _RUST_SHIM_AVAILABLE = True
+except Exception:  # pragma: no cover - exercised in pure-Python planner tests.
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)